### DISCUSSION TITLE
Windows Users: Share Your Setup - Hints & Tips for Other Members

### DISCUSSION THREAD

**Discussion by ms52538** [CodyBerenson/PGMA-Modernized#316] [General]
Windows Users:

Within the Plex Forums and elsewhere Plex User discussions are abound with questions about tweaking their configurations to help with improving performance for all things from Plex Media Server configurations, improving lookup performance, improving library performance, creation of Symbolic Links to extend coverage across multiple disk drives, etc.. It's a vast world out there - sometimes you have to dig for information - or perhaps members of PGAM-Modernized can help more specifically with information relative to the types of content we collect and stream.

While PGMA members concentrate on scraping and matching titles, there’s valuable knowledge to gain from sharing how each of us configure our Windows-based content environments. - each of us has a unique setup, including yours. Considering all aspects related to:

**YOUR ENVIRONMENT:**
1. your physical environment; ie. type of processor and ram used, disk types and config (i.e. using links), NAS and type, network switch, etc..
2. content acquisition (torrenting from private trackers or downloading from Internet-based sources),
3. content management through the Windows Explorer File System (including disk drive configurations like USB, internal, NAS HDDs, SSDs, and M.2 drives, along with the use of Symbolic Links), 
4. active maintenance of third-party applications (such as Torrent Clients), 
5. performance optimization for Plex Media Server (if any), 
6. library(s) configuration, 
7. and finally the tools and processes we employ to correctly name our content using PGMA-Modernized agents, or what we can do as workarounds when we can't match a title using PGMA.
 
Or perhaps you have **QUESTIONS** and would like answers from others:

- optimizing your disk/NAS performance and the overall performance of Plex Media Server, 
- size of your libraries (how many is too many?, how big can they get (# of files) before performance impacts?, 
- Scenes vs Features vs a Catch-all approach (beyond the points discussed in the PlexGayMetadataAgents-InstallationandUsageGuide.PDF), 
- how to organize titles that can be matched vs those that cannot be (i.e. OnlyFans-type content not listed in IAFD, specialty collections like Sports/Erotic Wrestling, private or amateur content, etc.), 
- performance options to improve operations, 
- how things such as Windows Search impacts your Plex-based environment (standard vs customized Plex Media Server setup, and considerations), 
- challenges you may encounter in managing multiple content ecosystems in tandem (i.e. Windows Explorer Sub-Directory setup, spread across drives, with an active Torrent client seeding and download from private trackers, running Plex Media Server).
- pro's vs con's of naming titles using correct formatting vs impact on Plex Search results 
- how to invite/friend members of this group to share your Plex environment with,
- and lastly possible apps/tools that can be used to help manage all/part of it.

In short, A LOT of information-gathering and planning, and trial & error goes into how we each setup our environments to run Plex Media Server and stream content. This post is meant to help foster sharing of information, tips & tricks, and apps/tools so that all of it can used to maximize your efforts based upon tried-and-true efforts of those who are members here within PGMA-Modernized.

Please feel free to loosely follow the formats outlined above and share your environments with the rest of us. If you have questions, then certainly feel free to ask and get input from the members here!

**Comment by Roganator** [comment]
Thank you for this thread, great idea!

Environment :
I have an old pc with an i3 processor and 4gb ram, which originally started as a seedbox but then grew into my plex server.  I originally had a KVM switch but now don't have a keyboard or monitor attached at all and instead just RDP into it.  My server is set to automatically log into the corresponding windows account and start Plex, and set in the bios to automatically restart in the event of a power outtage.  I also have a wifi-enabled power plug, so in short if it ever totally locks up or needs to be forcibly rebooted while I'm away from home I can just turn the power off and back on remotely.  Have only had to do a power cycle reboot twice in 7+ years, but it is nice to have.

Regarding content acquisition, gaytor is about all I use anymore.  It's also worth mentioning that while Plex started as a personal way to stream just adult videos, I now use it to stream all sorts of media such as movies and tv shows, and have shared access to various libraries with many friends and family.  I have a "Movies" library and "TV" library shared with family members, and even granularly use the 'labels' functionality to tag anime, lgbt, and other stuff - that way my friend's kid can just see anime using "allow only lables" under "restrictions", and my parents don't see anime or lgbt movies/tv shows via the "exclude labels" function.  I was initially concerned about having family-friendly media and porn all on the same server and all "in the same house" so to speak, but this approach has worked fine via a single plex server for me the last few years now.

Regarding content management through Windows Explorer File System, the bulk of my plex libraries are actually on USB external harddrives plugged directly into the server nowadays.  Yes, I know it's probably less than ideal but has worked fine for me.  While I was initially concerned they would be slower or less snappy... this hasn't caused any noticible performance or speed issues for me yet.  

I manually rename my files and don't have any tools or automated solutions in that regard ultimately, and can't think of any other helpful configuration or optimization tips...  I did use "tautulli" briefly when rolling out server access to friends and family to try and keep overall statistics of who was using it and how often... but it didn't ultimately seem to provide that much of an improvement over the built in dashboard ultimately.  There are a handful of Plex-specific facebook groups out there worth checking into, along with reddit.

I'm happy to help any way I can or answer any questions others may have about my specific setup, and also curious what other people have.

**Comment by jfreiman** [comment]
My genious tip is that I use Foobar 2000 (the music manager) to rename, move and enter Plex supported metadata into my collection.

I have modified much of Foobar 2000s GUI, the standard fields, and made it so I can use the added fields to rename the files with the data needed by PGMA.

I also can use foobar 2000 to find movies or scenes that contain 1 or more models -- something you can't do in Plex.

Here are a few of the changes I have made to Foobar 2000 for Plex

![image](https://github.com/user-attachments/assets/4b80fe79-0aa6-436a-8d18-bcfe693f27d5)

`DISPLAY

Autocomplete Fields
ALBUM TITLE;GENRE;ALBUM ARTIST;COPYRIGHT;UHD


PROPERTIES DIALOG

Field Name Suggestions:
ModelsPGMA,GEVI,IAFD,Copyright,artist,title,album,genre,tracknumber,composer,album artist

Multiline Fields
COMMENT;LYRICS;BIOGRAPHY;PERFORMER;ACTOR;ARTIST

Multivalue Fields
ARTIST;GENRE;ALBUM;ACTOR;DIRECTOR;GENRE;ALTTITLE;PERFORMER

Standard Fields:
-- PGMA --------------=pgma;Video Title (Track)=TITLE;Distributor (Album Artist)=ALBUM ARTIST;Year (Date)=DATE;Disc Number=DISCNUMBER;Total Discs=TOTALDISCS;Scene Number (Track)=TRACKNUMBER;Total Scenes (Tracks)=TOTALTRACKS;-- PLEX METADATA -----=plex;Genre (Safe, Bare, Pre-Condom)=GENRE;Plex Collections (Album Title)=ALBUM;Models Name (Artist)=ARTIST;PGMA [Cast]=ModelsPGMA;Plex Studio=COPYRIGHT;-- VIDEO  MANAGEMENT ---=mgt;Alternate Titles=ALTTITLE;Models Name (Hidden)=PERFORMER;Metadata URL=URL;Description (Comment)=COMMENT;Video Dimensions=DIMENSIONS;GEVI=GEVI
`
![image](https://github.com/user-attachments/assets/6b299ced-772d-447f-a194-ec69b7562fcb)

**Reply by jfreiman** [reply]
Foobar 2000 w/Plex PGMA

Batch attach Posters
![image](https://github.com/user-attachments/assets/179c1623-3724-42bf-9691-e659463d6cfb)

Batch Name videos from Filename/Title
![image](https://github.com/user-attachments/assets/751a5348-ed9c-46ff-bfc9-7653a8448c4d)

Fill in common data for all videos one time
![image](https://github.com/user-attachments/assets/d27411e6-b312-44e4-aa46-fb69407ffc83)

Use the filemane to add model names for scenes into metadata
![image](https://github.com/user-attachments/assets/6cf2b3d2-0f6c-4f46-8b7e-01d2ef096d4b)

Delete description from the performer fields
![image](https://github.com/user-attachments/assets/d41c5c2e-7b16-45f0-bd01-86898d2d9504)
![image](https://github.com/user-attachments/assets/e035530e-c123-4185-bfdf-01141f7533ce)
Use clean up to get rid of garbage
<img width="669" alt="image" src="https://github.com/user-attachments/assets/19f9e438-53b7-46ac-b197-aa7e129f0504">
![image](https://github.com/user-attachments/assets/95eeec09-2f4a-413f-a829-8b0e3cbf6e61)

Copy "performer" field to either the PGMA [Cast] (for adding to filename) or to Models Name (to input into video with Plex's built in metadata reader)
![image](https://github.com/user-attachments/assets/97b824cc-1000-4777-a8c5-fed08b589630)

Optionally add your own Plex Collection data (this supplements and does not overwrite PGMA) 
<img width="664" alt="image" src="https://github.com/user-attachments/assets/9a5e486f-e8a6-416d-a60a-be9cec3f299f">

If your video doens't have a match with PGMA (or your's is broken like mine is at the moment) you can use MP4 Video & Audio Tag Editor (Windows Store) to copy the description saved in Foobar 2000 into the "Long Description" field (which Foobar 2000 does not support) 
![image](https://github.com/user-attachments/assets/3dcb6ffa-47dc-4336-a558-2b257c82dff2)

then use Foobar 2000's File Move To operation to send your video to the proper folder in your Plex Collection -- I have separate move and copy options for Scenes, Features and others for convenience
![image](https://github.com/user-attachments/assets/e9e2a2fd-d530-40f7-80e9-ac247cd780da)

An example using the PGMA Scene agent.
<img width="667" alt="image" src="https://github.com/user-attachments/assets/0292dfad-12c5-4078-877d-096b5337dacc">

As you can see the GEVI # is added into the filemane
<img width="867" alt="image" src="https://github.com/user-attachments/assets/695e78cf-722d-4006-98e7-b319efbd1f4c">

The string I used is as follows
\%Album Artist%\['('%album artist%')'] - %title%[' - {'%GEVI%N'}'][' ('%date%')'][ - pt%track number%][ - '['%modelspgma%']']['[' - %Dimensions%']']

For Scenes
Destination folder: \\last8wide\adult scenes
File name pattern: \%Album Artist%\['('%album artist%')'] - %title%[' - {'%GEVI%N'}'][' ('%date%')'][ - pt%track number%][ - '['%modelspgma%']']['[' - %Dimensions%']']
![image](https://github.com/user-attachments/assets/2a8341e7-a805-41fd-8179-9136b611891b)

For Features
Destination folder: \\last8wide\adult features
File name pattern: \%Album Artist%\['('%album artist%')'] - %title% ['('%date%')'][ - '['%modelspgma%']'][ - Disk %discnumber%]['['%Dimensions%']']

**Comment by jfreiman** [comment]
One last example (for now)
here is how the PGMA [Cast] field works
<img width="872" alt="image" src="https://github.com/user-attachments/assets/a4447569-de97-47cd-a900-377e5aff8859">

**Comment by jfreiman** [comment]
After understanding that PGMA isn't currently I think my above setup using Foobar2000 and MP4 Video & Audio Tagger, it can help keep your adult libraries searchable using metadata by Copyright (Studio), Collection, Genre, Cast/Performers, the proper year (not added date), and Plex Summary if you use MP4 Video Tag editor.
The one problem introduced by using this technique to add _cast_ names is that it will mostly erase the image already stored in Plex by PGMA. You can sometimes get the 'image back' by going to a previously matched title and refreshing the web page (NOT Refresh Metadata!).

** Since I am hopeful that PGMA will be up and running again before Plex discontinues it's current support for Agents (I have zero inside information) I do one following change to the video metadata.

After the title(s) are added to Plex with the metadata I have inputted, I erase the "Model Name (Artist)" field so that when/if PGMA works again, it will be able to update and overwrite this field.
Plex Collections and Genre metadata is merged with PGMA's scrubbing and does not need to be removed for PGMA to update it.

**Comment by jfreiman** [comment]
I turned off the background image in Plex so that my page clipping doesn't have nudity in it, but it should give you a general idea of what you can expect.
![image](https://github.com/user-attachments/assets/19908665-b72c-4a3e-b400-b8e3fbe2bc82)
